cleanlab/outlier.py

```diff
@@ -0,0 +1,534 @@
+        if k > len(features):  # Ensure number of neighbors less than number of examples
+            raise ValueError(
+                f"Number of nearest neighbors k={k} cannot exceed the number of examples N={len(features)} passed into the estimator (knn)."
+            )
+        knn = NearestNeighbors(n_neighbors=k, metric="cosine").fit(features)
+        features = None  # features should be None in knn.kneighbors(features) to avoid counting duplicate data points
+    elif k is None:
+        k = knn.n_neighbors
+
+    max_k = knn.n_neighbors  # number of neighbors previously used in NearestNeighbors object
+    if k > max_k:  # if k provided is too high, use max possible number of nearest neighbors
+        warnings.warn(
+            f"Chosen k={k} cannot be greater than n_neighbors={max_k} which was used when fitting "
+            f"NearestNeighbors object! Value of k changed to k={max_k}.",
+            UserWarning,
+        )
+        k = max_k
+
+    # Fit knn estimator on the features if a non-fitted estimator is passed in
+    try:
+        knn.kneighbors(features)
+    except NotFittedError as e:
+        knn.fit(features)
+
+    # Get distances to k-nearest neighbors Note that the knn object contains the specification of distance metric
+    # and n_neighbors (k value) If our query set of features matches the training set used to fit knn, the nearest
+    # neighbor of each point is the point itself, at a distance of zero.
+    distances, _ = knn.kneighbors(features)
+
+    # Calculate average distance to k-nearest neighbors
+    avg_knn_distances = distances[:, :k].mean(axis=1)
+
+    # Map ood_features_scores to range 0-1 with 0 = most concerning
+    ood_features_scores: np.ndarray = np.exp(-1 * avg_knn_distances * t)
+    return (ood_features_scores, knn)
+
+
+def _get_ood_predictions_scores(
+    pred_probs: np.ndarray,
+    *,
+    labels: Optional[np.ndarray] = None,
+    confident_thresholds: Optional[np.ndarray] = None,
+    adjust_pred_probs: bool = True,
+    method: str = "entropy",
+) -> Tuple[np.ndarray, Optional[np.ndarray]]:
+    """Returns an OOD (out of distribution) score for each example based on it pred_prob values.
+
+    Parameters
+    ----------
+    pred_probs : np.ndarray
+      An array of shape ``(N, K)`` of model-predicted probabilities,
+      `pred_probs` in the same format expected by the :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+    confident_thresholds : np.ndarray, default = None
+      For details on 'confindent_thresholds` see :py:class:`OutOfDistribution <cleanlab.outlier.OutOfDistribution>`.
+
+    labels : np.ndarray, optional
+      `labels` in the same format expected by the :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
+
+    adjust_pred_probs : bool, True
+      Account for class imbalance in the label-quality scoring.
+      For details, see `adjust_pred_probs` in :py:func:`fit <cleanlab.outlier.OutOfDistribution.fit>` function.
```

Comment:
For details, see key `adjust_pred_probs` in the `params` dict arg of  cleanlab.outlier.OutOfDistribution class

(I presume you can figure out how to properly link the spot in docs for this class where this documentation appears)